posthog-node/src/posthog-node.ts

```diff
@@ -287,59 +287,80 @@ export class PostHog extends PostHogCoreStateless implements PostHogNodeV1 {
       distinctId,
       groups,
       personProperties,
       groupProperties
     )
 
-    personProperties = adjustedProperties.allPersonProperties
-    groupProperties = adjustedProperties.allGroupProperties
-
-    let response = undefined
-
-    // Try to get match value locally if not provided
-    if (!matchValue) {
+    if (matchValue === undefined) {
       matchValue = await this.getFeatureFlag(key, distinctId, {
         ...options,
         onlyEvaluateLocally: true,
+        sendFeatureFlagEvents: false,
       })
     }
 
-    if (matchValue) {
-      response = await this.featureFlagsPoller?.computeFeatureFlagPayloadLocally(key, matchValue)
-    }
+    let response: string | boolean | undefined
+    let payload: JsonType | undefined
 
-    // set defaults
-    if (onlyEvaluateLocally == undefined) {
-      onlyEvaluateLocally = false
-    }
-    if (sendFeatureFlagEvents == undefined) {
-      sendFeatureFlagEvents = true
+    if (matchValue) {
+      response = matchValue
+      payload = await this.featureFlagsPoller?.computeFeatureFlagPayloadLocally(key, matchValue)
+    } else {
+      response = undefined
+      payload = undefined
     }
 
-    // set defaults
-    if (onlyEvaluateLocally == undefined) {
-      onlyEvaluateLocally = false
-    }
+    // Determine if the payload was evaluated locally
+    const payloadWasLocallyEvaluated = payload !== undefined
 
-    const payloadWasLocallyEvaluated = response !== undefined
+    // Fetch final flags and payloads either locally or from the remote server
+    let fetchedOrLocalFlags: Record<string, string | boolean> | undefined
+    let fetchedOrLocalPayloads: Record<string, JsonType | undefined> | undefined
 
-    if (!payloadWasLocallyEvaluated && !onlyEvaluateLocally) {
-      response = await super.getFeatureFlagPayloadStateless(
-        key,
+    if (payloadWasLocallyEvaluated || onlyEvaluateLocally) {
+      if (response !== undefined) {
+        fetchedOrLocalFlags = { [key]: response }
+        fetchedOrLocalPayloads = { [key]: payload }
+      } else {
+        fetchedOrLocalFlags = {}
+        fetchedOrLocalPayloads = {}
+      }
+    } else {
+      const fetchedData = await super.getFeatureFlagsAndPayloadsStateless(
         distinctId,
         groups,
-        personProperties,
-        groupProperties,
+        allPersonProperties,
+        allGroupProperties,
         disableGeoip
       )
+      fetchedOrLocalFlags = fetchedData.flags || {}
+      fetchedOrLocalPayloads = fetchedData.payloads || {}
     }
-    return response
+
+    const finalResponse = fetchedOrLocalFlags[key]
+    const finalPayload = fetchedOrLocalPayloads[key]
+    const finalLocallyEvaluated = payloadWasLocallyEvaluated
+
+    this.capture({
+      distinctId,
+      event: '$feature_flag_called',
+      properties: {
+        $feature_flag: key,
+        $feature_flag_response: finalResponse,
+        $feature_flag_payload: finalPayload,
```

Comment:
is this flag only for local evaluation? I don't think we send this one for any other Mobile SDK, but we do support feature flag payloads

Reply:
What I meant is that I did not see this new property `feature_flag_payload` anywhere else yet, is that new? if so, we'd need to add to other SDKs as well?
JS for example has `$feature_flag_payloads` but its not used when capturing a `$feature_flag_called` event

Reply:
So my question about local evaluation is if this is only for backend SDKs (only backend SDKs have local evaluation afaik), because I have not found any reference on the JS SDK

Reply:
AH I see!  Okay this is a good point to raise: yes; right now this fallback on happens for local evaluation on backend SDKs, and we don't include any payload information in these events.  In fact, I don't even think we send `feature_flag_called` events whenever a client-side SDK is used to get payloads, the code just goes like this
```
    getFeatureFlagPayload(key: string): JsonType {
        const payloads = this.getFlagPayloads()
        return payloads[key]
    }
```
where `getFlagPayloads` just loads the payloads from persistence.
```
    getFlagPayloads(): Record<string, JsonType> {
        const flagPayloads = this.instance.get_property(PERSISTENCE_FEATURE_FLAG_PAYLOADS)
        return flagPayloads || {}
    }
```

So this is already feature gap between client-side and server side SDKs in more ways than just including the payload – our server-side SDKs send `feature_flag_called` events on both `getFeatureFlag` and `getFeatureFlagPayload` calls.  I feel like I should make this documentation clear somewhere if it isn't already, since users could get confused.

Reply:
yeah, I think that's a common pattern for the client-side SDKs that the server-side ones may not follow as much (e.g., the reason I started working on this change at all was because we have a user that's using our python SDK only for the `get_feature_flag_payload` method).

Okay, so here's what I'm going to do going forward:

- ship this change to the node SDK
- keep posthog-js as is

Thanks for chatting about it with me!